qemu/tests/migration_multi_host_with_xbzrle.py

```diff
@@ -0,0 +1,363 @@
+            transferred_ram_list.append(transferred_ram)
+            logging.info("The total time is %d, downtime is %d and "
+                         "transferred ram is %d after migration" %
+                         (total_time, downtime, transferred_ram))
+
+        @error.context_aware
+        def check_mig_totaltime_downtime_transferred_ram(self):
+
+            """
+            check total time, downtime and transferred ram after migration
+            the items in list should be decrease
+            """
+
+            if self.is_src:
+                error.context("Check total time, downtime and transferred ram"
+                              " after migration.", logging.info)
+                logging.info("Total time list: %s" % str(mig_total_time_list))
+                logging.info("Downtime list: %s" % str(mig_downtime_list))
+                logging.info("Transferred ram list: %s" %
+                             str(transferred_ram_list))
+                for i in range(len(mig_total_time_list)):
+                    if min(mig_total_time_list) != mig_total_time_list[-1]:
+                        raise error.TestFail("The total time of migration is "
+                                             "error, %s should be minimum, "
+                                             "but actual is %s" %
+                                             (mig_total_time_list[-1],
+                                              min(mig_total_time_list)))
+                    else:
+                        mig_total_time_list.pop()
+                    if min(mig_downtime_list) != mig_downtime_list[-1]:
+                        raise error.TestFail("The downtime of migration is "
+                                             "error, %s should be minimum, "
+                                             "but actual is %s" %
+                                             (mig_downtime_list[-1],
+                                              min(mig_downtime_list)))
+                    else:
+                        mig_downtime_list.pop()
+                    if min(transferred_ram_list) != transferred_ram_list[-1]:
+                        raise error.TestFail(
+                            "The transferred ram of migration is error, "
+                            "%s should be minimum, but actual is %s" %
+                            (transferred_ram_list[-1],
+                             min(transferred_ram_list)))
+                    else:
+                        transferred_ram_list.pop()
+
+        def before_migration_capability(self, mig_data):
+
+            """
+            get migration capability (xbzrle/cache_size)
+            enable/disable xbzrle
+
+            :param mig_data: Data for migration
+            """
+
+            if self.is_src:
+                self.get_migration_capability()
+                self.get_migration_cache_size()
+                if self.sub_test[0] == "set_xbzrle":
+                    self.set_xbzrle()
```

Comment:
Thanks vivianQizhu. In order to make the code easier to read and understand, add set_xbzrle and auto_converge. And it also avoids cfg file mistake.